src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -798,7 +812,82 @@ public SymbolInfo GetSpeculativeSymbolInfo(int position, ConstructorInitializerS
+        /// initializer that did not actually appear in the source code.
+        /// 
+        /// NOTE: This will only work in locations where there is already a constructor initializer.
+        /// </summary>
+        /// <param name="position">A character position used to identify a declaration scope and accessibility. This
+        /// character position must be within the span of an existing constructor initializer.
+        /// </param>
+        /// <param name="constructorInitializer">A syntax node that represents a parsed constructor initializer. This syntax node
+        /// need not and typically does not appear in the source code referred to SemanticModel instance.</param>
+        /// <returns>The semantic information for the topmost node of the constructor initializer.</returns>
+        public SymbolInfo GetSpeculativeSymbolInfo(int position, PrimaryConstructorBaseTypeSyntax constructorInitializer)
+        {
+            Debug.Assert(CanGetSemanticInfo(constructorInitializer, isSpeculative: true));
+
+            position = CheckAndAdjustPosition(position);
+
+            if (constructorInitializer == null)
+            {
+                throw new ArgumentNullException(nameof(constructorInitializer));
+            }
+
+            // NOTE: since we're going to be depending on a MemberModel to do the binding for us,
+            // we need to find a constructor initializer in the tree of this semantic model.
+            // NOTE: This approach will not allow speculative binding of a constructor initializer
+            // on a constructor that didn't formerly have one.
+            // TODO: Should we support positions that are not in existing constructor initializers?
+            // If so, we will need to build up the context that would otherwise be built up by
+            // InitializerMemberModel.
+            var existingConstructorInitializer = this.Root.FindToken(position).Parent.AncestorsAndSelf().OfType<PrimaryConstructorBaseTypeSyntax>().FirstOrDefault();
+
+            if (existingConstructorInitializer == null)
+            {
+                return SymbolInfo.None;
+            }
+
+            MemberSemanticModel memberModel = GetMemberModel(existingConstructorInitializer);
+
+            if (memberModel == null)
+            {
+                return SymbolInfo.None;
+            }
+
+            var argumentList = existingConstructorInitializer.ArgumentList;
+            var binder = memberModel.GetEnclosingBinder(LookupPosition.IsBetweenTokens(position, argumentList.OpenParenToken, argumentList.CloseParenToken) ? position : argumentList.OpenParenToken.SpanStart);
+            if (binder != null)
+            {
+                var diagnostics = DiagnosticBag.GetInstance();
+                binder = new ExecutableCodeBinder(constructorInitializer, binder.ContainingMemberOrLambda, binder);
+
+                BoundExpressionStatement bnode = binder.BindConstructorInitializer(constructorInitializer, diagnostics);
```

Comment:
I believe we need to do a nullable rewrite of these nodes.

Reply:
> I believe we need to do a nullable rewrite of these nodes.

It doesn't look like we do this for any GetSpeculativeXXX APIs. Please open an issue if you think they should do the rewrite.

---
In reply to: [444528944](https://github.com/dotnet/roslyn/pull/45351#discussion_r444528944) [](ancestors = 444528944)

Reply:
Added a note to [https://github.com/dotnet/roslyn/issues/35038](https://github.com/dotnet/roslyn/issues/35038).

---
In reply to: [444537180](https://github.com/dotnet/roslyn/pull/45351#discussion_r444537180) [](ancestors = 444537180,444528944)